lib/Sema/TypeCheckAttr.cpp

```diff
@@ -161,6 +161,8 @@ class AttributeChecker : public AttributeVisitor<AttributeChecker> {
   IGNORED_ATTR(BackDeployed)
   IGNORED_ATTR(Documentation)
   IGNORED_ATTR(LexicalLifetimes)
+  IGNORED_ATTR(NonEscapable)
+  IGNORED_ATTR(UnsafeNonEscapableResult)
```

Comment:
Instead of ignoring the attribute, I think it makes sense to add a diagnostic raising an error if the NonEscapableTypes feature is off, but one of these attributes was encountered.